examples/chat/client/tasks.nim

```diff
@@ -255,34 +295,36 @@ proc startWakuChat2*(username: string) {.task(kind=no_rts, stoppable=false).} =
 
   nick = username
 
-  when PayloadV1:
-    # generate `symKey` here instead of `statusContext` to decrease startup
-    # time of `TaskRunner` instance and therefore time to first paint of TUI
-    if not symKeyGenerated:
-      symKey = generateSymKey(contentTopic)
-      symKeyGenerated = true
-
-  let (extIp, extTcpPort, extUdpPort) = setupNat(conf.nat, clientId,
-    Port(uint16(conf.tcpPort) + conf.portsShift),
-    Port(uint16(conf.udpPort) + conf.portsShift))
-
-  var nodekey: waku_chat2.crypto.PrivateKey
-
-  if $conf.nodekey == "":
-    nodekey = waku_chat2.crypto.PrivateKey.random(Secp256k1,
-      waku_chat2.keys.newRng()[]).tryGet()
-  else:
-    nodekey = waku_chat2.crypto.PrivateKey(scheme: Secp256k1,
-      skkey: SkPrivateKey.init(
-        waku_chat2.utils.fromHex($conf.nodekey)).tryGet())
+  # invoke `setupNat` here instead of `statusContext` to decrease startup time
+  # of `TaskRunner` instance; doing so here also avoids e.g. possibly
+  # triggering an OS dialog re: permitting network activity until the time a
+  # waku node initially connects to the network
+  if not natIsSetup:
+    (extIp, extTcpPort, extUdpPort) = setupNat(conf.nat, clientId,
+      Port(uint16(conf.tcpPort) + conf.portsShift),
+      Port(uint16(conf.udpPort) + conf.portsShift))
+
+    natIsSetup = true
+
+  # generate `nodekey` here instead of `statusContext` to decrease startup
+  # time of `TaskRunner` instance and therefore time to first paint of TUI
+  if not nodekeyGenerated:
+    if $conf.nodekey == "":
+      nodekey = waku_chat2.crypto.PrivateKey.random(Secp256k1,
+        waku_chat2.keys.newRng()[]).tryGet()
+    else:
+      nodekey = waku_chat2.crypto.PrivateKey(scheme: Secp256k1,
+        skkey: SkPrivateKey.init(
+          waku_chat2.utils.fromHex($conf.nodekey)).tryGet())
+
+    nodekeyGenerated = true
 
   wakuNode = WakuNode.init(nodekey, ValidIpAddress.init($conf.listenAddress),
     Port(uint16(conf.tcpPort) + conf.portsShift), extIp, extTcpPort)
 
   await wakuNode.start()
 
-  wakuNode.mountRelay(conf.topics.split(" "),
-    rlnRelayEnabled = conf.rlnRelay,
+  wakuNode.mountRelay(conf.topics.split(" "), rlnRelayEnabled = conf.rlnRelay,
```

Comment:
Again, why aren't we using the topics in context?

Reply:
waku v2 `topic`s are distinct from `contentTopic`s. See: https://rfc.vac.dev/spec/23/.